cmake/SpectreCheckTargetDependencies.cmake

```diff
@@ -371,6 +371,21 @@ function(_check_and_print_dependencies
   cmake_parse_arguments(
     ARG "ERROR_ON_FAILURE" "" "" ${ARGN})
 
+  # We add the dependencies of allowed targets as additional allowed targets.
+  # This avoids the situation where target A is allowed, but A depends on B,
+  # and therefore every target has a dependency on B.
+  set(LIST_OF_ALLOWED_EXTRA_TARGETS_AND_DEPS
+    ${LIST_OF_ALLOWED_EXTRA_TARGETS})
+  foreach(_ALLOWED_EXTRA_TARGET ${LIST_OF_ALLOWED_EXTRA_TARGETS})
```

Comment:
question: wouldn't you want this to be more recursive?As in, SpectreAllocatior depends on Jemalloc, which depends on jemalloc.so, so we would want to add `SpectreAllocator` explicitly, then have this implicitly add `Jemalloc` and `jemalloc.so`

Reply:
Agreed, I'll fix